django/contrib/postgres/search.py

```diff
@@ -92,14 +92,38 @@ def __init__(self, lhs, connector, rhs, config, output_field=None):
         super(CombinedSearchVector, self).__init__(lhs, connector, rhs, output_field)
 
 
-class SearchQuery(Value):
+class SearchQueryCombinable(object):
+    BITAND = '&&'
+    BITOR = '||'
+
+    def _combine(self, other, connector, reversed, node=None):
+        if not isinstance(other, SearchQueryCombinable) or not self.config == other.config:
+            raise TypeError('SearchQuery can only be combined with other SearchQuerys')
+        if reversed:
+            return CombinedSearchQuery(other, connector, self, self.config)
+        return CombinedSearchQuery(self, connector, other, self.config)
+
+    # On Combinable, these are not implemented to reduce confusion with Q. In
+    # this case we are actually (ab)using them to do logical combination so
+    # it's consistent with other usage in Django.
+    def __or__(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __ror__(self, other):
```

Comment:
Combining with an object that doesn't have `__or__` or `__and__` defined, for example `None`, it raises a better error message.
That's the only test case I have found, but I think it's very useful to get `TypeError: SearchQuery can only be combined with other SearchQuerys` instead of `NotImplementedError: Use .bitand() and .bitor() for bitwise logical operations.`